docs/.vuepress/theme/components/Navigation.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+  <div class="top-wrapper">
+    <router-link to="/">
+      <img
+        class="header-logo"
+        src="../images/ethereum-logo-wireframe.png"
+        alt="Ethereum Logo"
+      />
+    </router-link>
+    <NavLinks
+      :isDarkMode="isDarkMode"
+      :isMobileNavVisible="isMobileNavVisible"
+      :handleNavToggle="handleNavToggle"
+      @nav-toggle="handleNavToggle"
+      @dark-mode-toggle="$emit('dark-mode-toggle')"
+    />
+
+    <span class="icon-link icon-menu" tabindex="0">
+      <icon
+        name="menu"
+        class="icon-menu"
+        @click.native="handleNavToggle(true)"
+      />
+    </span>
+  </div>
+</template>
+
+<script>
+import NavLinks from './NavLinks.vue'
+import SearchBox from './SearchBox.vue'
+
+export default {
+  components: { NavLinks },
+  props: {
+    isDarkMode: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data() {
+    return {
+      isMobileNavVisible: false
+    }
+  },
+  computed: {
+    darkOrLightModeIcon() {
+      return this.isDarkMode ? 'darkmode' : 'lightmode'
+    }
+  },
+  methods: {
+    handleNavToggle(value) {
```

Comment:
I believe this can be simplified. Since this is merely toggling a boolean, we don't need to pass in a value. Test changing this to:
```
    handleNavToggle() {
      this.isMobileNavVisible = !this.isMobileNavVisible
    }
```

Reply:
Then any instances of emitting the toggle can be updated from:
```$emit('nav-toggle', false)```
To
```$emit('nav-toggle')```

Reply:
Certain buttons only do specific things, so I'm averse to toggling the value. We'd have to then likely have to monitor the viewport width  to prevent anything clashing as a result of the nav toggling
- When a nav link is clicked, the nav should always hide
- When the close icon is clicked, the nav should always hide
- When the menu icon is clicked, the nav should always open

Reply:
Right but the nav links & close icon are only visible when the nav is visible & the menu icon is only visible when the nav is closed. So the toggling boolean logic works, no viewport monitoring necessary 😃

Reply:
Just to return to this conversation, not being explicit here toggles the overflow on desktop too, and to keep things simple I've reintroduced the bool. Not having this would prevent the user from scrolling